python/semantic_kernel/connectors/ai/open_ai/prompt_execution_settings/open_ai_prompt_execution_settings.py

```diff
@@ -41,16 +41,19 @@ class OpenAITextPromptExecutionSettings(OpenAIPromptExecutionSettings):
     @model_validator(mode="after")
     def check_best_of_and_n(self) -> "OpenAITextPromptExecutionSettings":
         """Check that the best_of parameter is not greater than the number_of_responses parameter."""
-        if self.best_of is not None and self.best_of < self.number_of_responses:
-            raise ServiceInvalidExecutionSettingsError(
-                "When used with number_of_responses, best_of controls the number of candidate completions and n specifies how many to return, therefore best_of must be greater than number_of_responses."  # noqa: E501
-            )
-        if self.extension_data.get("best_of") is not None and self.extension_data["best_of"] < self.extension_data.get(
-            "number_of_responses"
+        if self.best_of is not None and self.number_of_responses is not None:
+            if self.best_of < self.number_of_responses:
```

Comment:
this can be done in one if statement... as long as you check the is not None first, it is just as fast and reduces code